tests/vmi_multus_test.go

```diff
@@ -873,6 +875,48 @@ var _ = Describe("SRIOV", func() {
 		It("[test_id:3957]should connect to another machine with sriov interface over IPv6", func() {
 			pingThroughSriov("fc00::1/64", "fc00::2/64")
 		})
+
+		Context("VLANed SR-IOV network", func() {
+			var vlanedVMI1 *v1.VirtualMachineInstance
+			cidrVlaned1 := "192.168.0.1/24"
+			cidrVlaned2 := "192.168.0.2/24"
+			cidrNonVlaned := "192.168.0.3/24"
+			tests.BeforeAll(func() {
+				createNetwork("sriov-vlan", tests.NamespaceTestDefault, sriovConfVlanCRD)
+				createNetwork("sriov-no-vlan", tests.NamespaceTestDefault, sriovConfNoVlanCRD)
+			})
+
+			BeforeEach(func() {
+				vlanedVMI1 = createSriovVMI([]string{"sriov-vlan"})
+
+				// manually configure IP/link on sriov interfaces because there is
+				// no DHCP server to serve the address to the guest
+				configInterface(vlanedVMI1, "eth1", cidrVlaned1, "#")
+			})
+
+			AfterEach(func() {
+				deleteNetwork("sriov-vlan", tests.NamespaceTestDefault)
```

Comment:
Now the reation is in the`BeforeAll`, I don't think the removal should be `AfterEach`

Reply:
actually I have second thoughts about the beforeall() as well. because we loose tests isolation. please take a look at this [explanation](https://github.com/onsi/ginkgo/issues/70#issuecomment-42882976), and also @maiqueb  referred to it some time ago.
what do you think?

Reply:
Thanks for the link:) AFAIU, `NetworkAttachmentDefinition` is stateless (unless a test will manually change it, reusing it shouldn't affect it), so it should be fine to share it between the `it`s. But I understand your case.
I'm ok with having it in the `BeforeEach` or `BeforeAll`. If you choose the `BeforeAll` then you should remove the network deletion in `AfterEach` (otherwise just the first test in the `Context` will work).

Reply:
so I prefer to move it back to the beforeEach() :)